js/style/style_batch.js

```diff
@@ -46,11 +48,22 @@ styleBatch.prototype = {
         if (this._style._layers[layer.id] !== undefined) {
             throw new Error('There is already a layer with this ID');
         }
+
+
         if (!(layer instanceof StyleLayer)) {
+            if (validateStyle.emitErrors(this._style, validateStyle.layer({
```

Comment:
Is there a compelling reason to take on this technical debt now? E.g. is this blocking an important release?

I'd like to at least have a nicer API exposed by the local `./validate_style` even if we don't push it up to mbgl-style-spec yet.